celery/concurrency/base.py

```diff
@@ -44,6 +45,9 @@ def apply_target(target, args=(), kwargs={}, callback=None,
     else:
         callback(ret)
 
+    # Switch to the tornado ioloop greenlet
+    future_executor.switch_to_ioloop()
```

Comment:
Pretty sure this doesn't belong here since tornado workers pool should be optional.